src/nupic/engine/RegionImplFactory.cpp

```diff
@@ -297,64 +299,90 @@ static RegionImpl * deserializePyNode(DynamicPythonLibrary * pyLib,
 
 }
 
-RegionImpl* RegionImplFactory::createRegionImpl(const std::string nodeType, 
+RegionImpl* RegionImplFactory::createRegionImpl(const std::string nodeType,
                                                 const std::string nodeParams,
                                                 Region* region)
 {
 
-  RegionImpl *mn = nullptr;
+  RegionImpl *impl = nullptr;
   Spec *ns = getSpec(nodeType);
   ValueMap vm = YAMLUtils::toValueMap(
-    nodeParams.c_str(), 
-    ns->parameters, 
-    nodeType, 
+    nodeParams.c_str(),
+    ns->parameters,
+    nodeType,
     region->getName());
-    
+
   if (cppRegions.find(nodeType) != cppRegions.end())
   {
-    mn = cppRegions[nodeType]->createRegionImpl(vm, region);
+    impl = cppRegions[nodeType]->createRegionImpl(vm, region);
   }
   else if ((nodeType.find(std::string("py.")) == 0))
   {
     if (!pyLib_)
       pyLib_ = boost::shared_ptr<DynamicPythonLibrary>(new DynamicPythonLibrary());
-    
-    mn = createPyNode(pyLib_.get(), nodeType, &vm, region);
+
+    impl = createPyNode(pyLib_.get(), nodeType, &vm, region);
   } else
   {
     NTA_THROW << "Unsupported node type '" << nodeType << "'";
   }
 
-  return mn;
+  return impl;
 
 }
 
-RegionImpl* RegionImplFactory::deserializeRegionImpl(const std::string nodeType, 
+RegionImpl* RegionImplFactory::deserializeRegionImpl(const std::string nodeType,
                                                      BundleIO& bundle,
                                                      Region* region)
 {
 
-  RegionImpl *mn = nullptr;
+  RegionImpl *impl = nullptr;
 
   if (cppRegions.find(nodeType) != cppRegions.end())
   {
-    mn = cppRegions[nodeType]->deserializeRegionImpl(bundle, region);
+    impl = cppRegions[nodeType]->deserializeRegionImpl(bundle, region);
   }
   else if (StringUtils::startsWith(nodeType, "py."))
   {
     if (!pyLib_)
       pyLib_ = boost::shared_ptr<DynamicPythonLibrary>(new DynamicPythonLibrary());
-    
-    mn = deserializePyNode(pyLib_.get(), nodeType, bundle, region);
+
+    impl = deserializePyNode(pyLib_.get(), nodeType, bundle, region);
   } else
   {
     NTA_THROW << "Unsupported node type '" << nodeType << "'";
   }
-  return mn;
+  return impl;
 
 }
 
-// This function returns the node spec of a NuPIC 2 or NuPIC 1 Python node 
+RegionImpl* RegionImplFactory::deserializeRegionImpl(
+    const std::string nodeType,
+    capnp::AnyPointer::Reader& proto,
+    Region* region)
+{
+  RegionImpl *impl = nullptr;
+
+  if (cppRegions.find(nodeType) != cppRegions.end())
+  {
+    impl = cppRegions[nodeType]->deserializeRegionImpl(proto, region);
+  }
+  else if (StringUtils::startsWith(nodeType, "py."))
+  {
+    NTA_THROW << "Python regions not yet supported for Cap'n Proto "
+      << "deserialization.";
+    //if (!pyLib_)
+    //  pyLib_ = boost::shared_ptr<DynamicPythonLibrary>(new DynamicPythonLibrary());
+
+    //impl = deserializePyNode(pyLib_.get(), nodeType, proto, region);
```

Comment:
This code will work, as is, once the PyRegion C++ class defined in NuPIC implements the new read/write functions. Can I put a comment stating this and leave the code in?